pkg/raft/testdata/slow_follower_after_compaction.txt

```diff
@@ -96,24 +96,52 @@ compact 1 17
 ----
 1/18 EntryNormal "prop_1_18"
 
-# Trigger a round of empty MsgApp "probe" from leader. It will reach node 3
-# which will reply with a rejection MsgApp because it sees a gap in the log.
-# Node 1 will reset the MsgApp flow and send a snapshot to catch node 3 up.
+# Trigger a round of MsgHeartbeat from the leader. This will cause the leader
+# to first call maybeSendAppend() which will realize that the log got truncated
+# and that follower 3 would reject the MsgApp. Therefore, the leader will send
```

Comment:
It's not that the follower would necessarily reject the `MsgApp`. We can't definitively say that about any `MsgApp` because the leader's idea about the follower's log is always stale. Even if we send a random slice of the log (irrespective of the `Next` index), there is a chance the follower magically caught up from some other source (e.g. a stale message from this leader finally got through to it) and accepts this `MsgApp`.

But the reason we send a snapshot is simply that we can't construct a `MsgApp` starting at the `Next` index, and that disrupts the `StateReplicate` "algorithm" - always be sending the log contiguously.

Suggestion:

```
This causes the leader to attempt sending a MsgApp to the follower 3. Since
the log got truncated beyond the Next index, the leader can't stay in
StateReplicate, and sends a snapshot instead.
```